Remove Reader Improvements feature flag

malinajirka opened the pull request:
Fixes #12026 

This PR removes the Reader Improvements Phase 2 feature flag -> it enables the feature on all builds.

Merge instructions
1. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/12866 is merged
2. Remove "Not Ready for Merge" label

Please review each line I changed very carefully in this PR - it's quite easy to miss `!` or something similar and end up with a wrong condition. Thanks!

To test:
1. Smoke test Reader - make sure the new discover is shown

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

malinajirka committed:
Remove Reader Improvements feature flag

malinajirka committed:
Update RELEASE-NOTES.txt

malinajirka commented on RELEASE-NOTES.txt:
```diff
@@ -1,6 +1,8 @@
 15.7
 -----
- 
+ * [***] Reader: Introduced a new Discover tab tailored for each user.
```
Any suggestions for the release notes are very appreciated :).

malinajirka committed:
Fix lint

malinajirka committed:
Fix lint

ashiagr commented on WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java:
```diff
@@ -311,7 +311,7 @@ private void renderTagHeader(
                 ReaderTagTable.isFollowedTagName(currentTag.getTagSlug()),
                 isFollowButtonEnabled,
-                AppPrefs.isReaderImprovementsPhase2Enabled() || mAccountStore.hasAccessToken()
+                mAccountStore.hasAccessToken()
```
Tag header changes were released before RI FF flag was removed [here](https://github.com/wordpress-mobile/WordPress-Android/pull/12689). At that time we did not allow following tags locally for the logged out users. We can remove this access token check now.

malinajirka replied:
Ohh nice catch! Thanks 🙇 

Fixed in dfd1c7101cfb52ffe782f0ec1f9aa32fe1f93f19

malinajirka committed:
Always show follow button on tag detail screen

malinajirka committed:
Merge branch 'develop' into issue/12026-remove-ri-feature-flag

# Conflicts:
#	WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderFragment.kt
#	WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java
#	WordPress/src/main/java/org/wordpress/android/ui/reader/services/update/ReaderUpdateLogic.java

ashiagr committed:
Fix logged out behaviour for tags action

1. Separate deleteTags into two methods
a. local only
b. local + remote

2. Pass result to action listener for local only action

ashiagr committed:
Pass logged-in status to delete tag action

ashiagr committed:
Revert "Fix logged out behaviour for tags action"

This reverts commit 8af8ef65

ashiagr committed:
Revert "Pass logged-in status to delete tag action"

This reverts commit 66f36b17

ashiagr commented:
Thanks @malinajirka ! 
Changes look good to me 👍.

While testing I found a bug unrelated to these changes regarding follow/unfollow tags in a logged out scenario. Fixed it here: #12875.
